app/mailers/publish_mailer.rb

```diff
@@ -1,20 +1,19 @@
 # frozen_string_literal: true
 
 class PublishMailer < ApplicationMailer
-  self.delivery_job = EmailDeliveryJob
+  helper :scheduling, :edition_url
 
-  add_template_helper(EditionUrlHelper)
+  self.delivery_job = EmailDeliveryJob
 
-  def publish_email(edition, user)
-    @user = user
+  def publish_email(recipient, edition, status)
     @edition = edition
-    @status = edition.status
+    @status = status
 
-    unless @status.published? || @status.published_but_needs_2i?
-      raise "Cannot send publish email with a non-published state"
+    if !status.published? && !status.published_but_needs_2i?
```

Comment:
Minor, but I'm surprised the linter accepts this, since the previous version was more readable to me. How come it changed?

Reply:
Heh, didn't realise this was a controversial one. With [some](http://www.railstips.org/blog/archives/2008/12/01/unless-the-abused-ruby-conditional/) [googling](https://github.com/rubocop-hq/rubocop/issues/5388) I realise it's a point of controversy in the ruby community.

But yeah I thought the linter might have ruled that the previous one was bad.

To me the `unless` reads wrong once you have more than one condition as it reads as "unless published or published but needs 2i" which makes the second clause ambiguous. Whereas "if not published and not published but needs 2i" you lack that ambiguity so the code is more readable.

I'll change it back though since it's not unanimous.